tests/Aql/ExecutionBlockImplTest.cpp

```diff
@@ -391,6 +401,1167 @@ TEST_F(ExecutionBlockImplTest,
+   *   2. result.length = (hardLimit||data.length) - offset.
+   *   3. result register entry matches the entry at the correct position in data.
+   *
+   * @param data The data to be expected, if we would just get it in full
+   * @param skipped The number of rows the executor reported as skipped
+   * @param result The resulting data output
+   * @param testReg The register to evaluate
+   */
+  void ValidateResult(std::shared_ptr<VPackBuilder> data, size_t skipped,
+                      SharedAqlItemBlockPtr result, RegisterId testReg) {
+    auto const& call = GetParam();
+
+    TRI_ASSERT(data != nullptr);
+    TRI_ASSERT(data->slice().isArray());
+
+    VPackSlice expected = data->slice();
+    VPackArrayIterator expectedIt{expected};
+    // Skip Part
+    size_t offset =
+        (std::min)(call.getOffset(), static_cast<size_t>(expected.length()));
+
+    if (!call.needsFullCount()) {
+      // Otherweise skipped = offset + fullCount
+      EXPECT_EQ(offset, skipped);
+    }
+
+    for (size_t i = 0; i < offset; ++i) {
+      // The first have been skipped
+      expectedIt++;
+    }
+    size_t limit =
+        (std::min)(call.getLimit(), static_cast<size_t>(expected.length()) - offset);
+    if (result != nullptr) {
+      // GetSome part
+      EXPECT_EQ(limit, result->size());
+      for (size_t i = 0; i < limit; ++i) {
+        // The next have to match
+        auto got = result->getValueReference(i, testReg).slice();
+        EXPECT_TRUE(basics::VelocyPackHelper::equal(got, *expectedIt, false))
+            << "Expected: " << expectedIt.value().toJson() << " got: " << got.toJson()
+            << " in row " << i << " and register " << testReg;
+        expectedIt++;
+      }
+    } else {
+      EXPECT_EQ(limit, 0);
+    }
+
+    // Now test Fullcount
+    if (call.needsFullCount()) {
+      ASSERT_TRUE(expected.length() >= offset + limit);
+      size_t fullCount = expected.length() - offset - limit;
+      EXPECT_EQ(offset + fullCount, skipped);
+    }
+  }
+};
+
+// Test a simple produce block. that has is supposed to write 1000 rows.
+TEST_P(ExecutionBlockImplExecuteIntegrationTest, test_produce_only) {
+  auto singleton = createSingleton();
+
+  auto builder = std::make_shared<VPackBuilder>();
+  builder->openArray();
+  for (size_t i = 0; i < 1000; ++i) {
+    builder->add(VPackValue(i));
+  }
+  builder->close();
+  RegisterId outReg = 0;
+  auto producer = produceBlock(singleton.get(), builder, outReg);
+
+  auto const& call = GetParam();
+  AqlCallStack stack{call};
+  auto const [state, skipped, block] = producer->execute(stack);
+  if (std::holds_alternative<size_t>(call.softLimit) && !call.hasHardLimit()) {
+    EXPECT_EQ(state, ExecutionState::HASMORE);
+  } else {
+    EXPECT_EQ(state, ExecutionState::DONE);
+  }
+
+  ValidateResult(builder, skipped, block, outReg);
+}
+
+// Test two consecutive produce blocks.
+// The first writes 10 lines
+// The second another 100 per input (1000 in total)
```

Comment:
The comment seems misleading, doesn't it have to be
```suggestion
// The second another 10 per input (100 in total)
```
instead?